packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -400,14 +427,44 @@ export class ContextsManager {
           console.error(`informer connect error on path ${path} for context ${context.name}: `, String(err));
           options.onConnectionError?.(String(err));
         }
-        options.onReachable?.(err === undefined);
+        if (err) {
+          this.setReachableNow(options, false);
+        } else {
+          this.setReachableDelay(options, true);
+        }
       });
     }
     this.restartInformer<T>(informer, context, options);
     return informer;
   }
 
-  private restartInformer<T>(
+  private setReachableDelay<T extends KubernetesObject>(options: CreateInformerOptions<T>, reachable: boolean): void {
+    clearTimeout(options.connectionDelay);
+    options.connectionDelay = setTimeout(() => {
+      this.doSetReachable(options, reachable);
+    }, connectTimeout);
+  }
+
+  private setReachableNow<T extends KubernetesObject>(options: CreateInformerOptions<T>, reachable: boolean): void {
+    clearTimeout(options.connectionDelay);
+    options.connectionDelay = undefined;
+    this.doSetReachable(options, reachable);
+  }
```

Comment:
should it be a common method ?
also, would it be important if now would just call with a timeout of 0

Reply:
I changed to always use a timeout, either > 0 or 0

Reply:
thanks